exchange-contract/exchange/contracts/issuer.cpp

```diff
@@ -166,14 +166,18 @@ bool ww::exchange::issuer::transfer(const Message& msg, const Environment& env,
     const std::string new_owner(msg.get_string("new_owner_identity"));
     ASSERT_SUCCESS(rsp, new_owner.size() > 0, "invalid transfer request, invalid owner identity parameter");
 
+    // if the old and new accounts are the same, then there is nothing to be done
+    if (old_owner == new_owner)
+        return rsp.success(false);
+
     const int count = (int) msg.get_number("count");
     ASSERT_SUCCESS(rsp, count > 0, "invalid transfer request, invalid asset count");
 
     // if there is no issuance for this identity, we treat it as a 0 balance
     ww::exchange::LedgerEntry old_entry;
     ASSERT_SUCCESS(rsp, ledger_store.get_entry(old_owner, old_entry),
                    "transfer failed, insufficient balance for transfer");
-    ASSERT_SUCCESS(rsp, count < old_entry.asset_.count_,
+    ASSERT_SUCCESS(rsp, count <= old_entry.asset_.count_,
                    "transfer failed, insufficient balance for transfer");
```

Comment:
Given that we permit transferring all "counts" of the old owner to the new owner (and thus making asset_.count = 0 for the old owner), wondering why in line 89 (as part of the issue method), we do not permit issuing "zero" counts. I would imagine this is perhaps equivalent to creating a bank account for someone with "zero balance"? I guess the real question is "is there a notion of bank for the transfer to happen?" or can the transfer method be done on any identities?

Reply:
i'm not sure i understand the question... semantically, we assume that there is an entry for every identity though for most the balance is 0 (there is no notion of "create an account"). however, MOST 0 balance accounts are not explicitly tracked. 

this specific fix allows for all funds from an account to be transferred. we don't explicitly delete the account (this is one of the cases where an account may have a 0 balance).